lib/features/recurring_donations/overview/widgets/recurring_donation_item.dart

```diff
@@ -0,0 +1,196 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:givt_app/core/enums/country.dart';
+import 'package:givt_app/features/auth/cubit/auth_cubit.dart';
+import 'package:givt_app/features/recurring_donations/overview/models/recurring_donation.dart';
+import 'package:givt_app/utils/app_theme.dart';
+import 'package:givt_app/utils/util.dart';
+import 'package:intl/intl.dart';
+
+class RecurringDonationItem extends StatefulWidget {
+  const RecurringDonationItem({
+    required this.recurringDonation,
+    required this.isExtended,
+    required this.onTap,
+    required this.onCancel,
+    super.key,
+  });
+
+  final RecurringDonation recurringDonation;
+  final bool isExtended;
+  final void Function() onTap;
+  final void Function() onCancel;
+  @override
+  State<RecurringDonationItem> createState() => _RecurringDonationItemState();
+}
+
+class _RecurringDonationItemState extends State<RecurringDonationItem> {
+  //TODO: POEditor
+  final frequencies = ['week', 'month', 'quarter', 'half year', 'year'];
+
+  String get frequencyText {
+    final user = context.read<AuthCubit>().state.user;
+    final currency = NumberFormat.simpleCurrency(
+      name: Util.getCurrencyName(country: Country.fromCode(user.country)),
+    );
+
+    //TODO: POEditor
+    final result =
+        'Each ${frequencies[widget.recurringDonation.frequency]} you give ${currency.currencySymbol} ${widget.recurringDonation.amountPerTurn.toStringAsFixed(2)}';
+
+    return result;
+  }
+
+  String get endsOnText {
+    final dateFormat = DateFormat('dd-MM-yyyy');
+
+    //TODO: POEditor
+    final endsOn =
+        'This will stop on ${dateFormat.format(widget.recurringDonation.endDate)}';
+    return endsOn;
+  }
+
+  @override
+  Widget build(BuildContext context) {
```

Comment:
It looks good but if we can increase the readability by splitting it into different methods or reusable widgets (we we have to reuse them)